block/retriever.go

```diff
@@ -99,19 +99,54 @@ func (m *Manager) processNextDABatch(ctx context.Context, daHeight uint64) error
 	return nil
 }
 
-func (m *Manager) fetchBatch(daHeight uint64) (da.ResultRetrieveBatch, error) {
+func (m *Manager) fetchBatch(daMetaData *da.DAMetaData) (da.ResultRetrieveBatch, error) {
 	var err error
-	batchRes := m.retriever.RetrieveBatches(daHeight)
+	availRes := m.retriever.CheckBatchAvailability(daMetaData)
+
+	switch availRes.Code {
+	case da.StatusUnableToGetProof:
+		//Is not possible to obtain proofs for the specific commitment in the specific height.
+		//This means there is no matching blob for that commitment
+		err = fmt.Errorf("Unable to get proof on height %d for the commitment", daMetaData.Height)
+
+	case da.StatusProofNotMatching:
+		//The proofs are obtained for the commitment, but not matching with the span (index, length) committed to the Hub.
+		err = fmt.Errorf("Span not matching the commitment")
+	case da.StatusError:
+		//There's been an issue validating proofs
+		err = fmt.Errorf("Error validating batch")
+	}
+	if !availRes.DataAvailable {
+		err = fmt.Errorf("no batches found on height %d", daMetaData.Height)
+		//There is no point on fetching the data
+		batchRes := da.ResultRetrieveBatch{}
```

Comment:
it's better to just insert the fields in the ResultRetrieveBatch while you create it vs adding it after